lib/vm.js

```diff
@@ -286,6 +287,72 @@ function runInThisContext(code, options) {
   return createScript(code, options).runInThisContext(options);
 }
 
+function compileFunction(code, params, options) {
+  const defaults = {
```

Comment:
This object should be lifted out of the function as it's never changed. Optionally it can also be `Object.freeze()`d.

Reply:
Will do.

Reply:
Of course, `defaults` object won't be necessary if you instead use destructuring with defaults as I did in https://github.com/nodejs/node/pull/21571#discussion_r200185676.

Reply:
That's exactly what @targos suggested, but then I'm also using the same object for type-checking. Would transition to a hard-coded types approach instead of the inferred approach as used here, it sounds better in this case anyway.

Reply:
> Would transition to a hard-coded types approach instead of the inferred approach as used here

Yes that is exactly what I'm suggesting in https://github.com/nodejs/node/pull/21571#discussion_r200185676.